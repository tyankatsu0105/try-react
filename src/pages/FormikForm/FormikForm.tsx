import React, { FunctionComponent } from 'react';
import { Helmet } from 'react-helmet';
import { Formik, Field, Form, ErrorMessage } from 'formik';
// import { render } from '@testing-library/react';
import * as Yup from 'yup';

const initialValues = {
  user: 'tyankatsu',
  email: 'hogehoge@fugafuga.piyo',
  social: {
    facebook: 'tyankatsu-facebook',
    twitter: 'tyankatsu-twitter',
  },
};

const validationSchema = Yup.object().shape({
  user: Yup.string()
    .min(2, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
  email: Yup.string()
    .min(2, 'Too Short!')
    .max(50, 'Too Long!')
    .email('lllllll')
    .required('Required'),
});

const FormikForm: FunctionComponent = () => {
  return (
    <>
      <Helmet>
        <title>This is FormikForm</title>
      </Helmet>

      <Formik
        initialValues={initialValues}
        onSubmit={(values) => {
          console.log(values);
        }}
        validationSchema={validationSchema}
        render={({ status, isSubmitting }) => (
          <Form>
            <Field type="email" name="email" />
            <ErrorMessage name="email" component="div" />
            <Field type="text" className="error" name="social.facebook" />
            <ErrorMessage name="social.facebook">
              {(errorMessage) => <div className="error">{errorMessage}</div>}
            </ErrorMessage>
            <Field type="text" name="social.twitter" />
            <ErrorMessage
              name="social.twitter"
              className="error"
              component="div"
            />
            {status && status.msg && <div>{status.msg}</div>}
            <button type="submit" disabled={isSubmitting}>
              Submit
            </button>
          </Form>
        )}
      ></Formik>
    </>
  );
};

export default FormikForm;
