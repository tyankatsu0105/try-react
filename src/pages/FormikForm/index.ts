import { lazy } from 'react';

export const FormikForm = lazy(() => import('./FormikForm'));
