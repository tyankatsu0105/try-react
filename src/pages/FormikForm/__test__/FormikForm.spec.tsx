import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import FormikForm from '../FormikForm';

describe('FormikForm', () => {
  it('snapshot', () => {
    const { baseElement } = render(
      <MemoryRouter initialEntries={['/']}>
        <FormikForm />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
