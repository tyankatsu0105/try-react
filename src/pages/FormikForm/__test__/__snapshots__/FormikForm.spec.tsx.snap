// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikForm snapshot 1`] = `
<body>
  <div>
    <form>
      <input
        name="email"
        type="email"
        value="hogehoge@fugafuga.piyo"
      />
      <input
        class="error"
        name="social.facebook"
        type="text"
        value="tyankatsu-facebook"
      />
      <input
        name="social.twitter"
        type="text"
        value="tyankatsu-twitter"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</body>
`;
